### ISSUE TITLE
Menu driven everything

### ISSUE THREAD

**Issue by matthewkastor** [design-first/system-designer#117]
This is a feature request to create a visual edit mode for defining schema. Something like a 2 column table with the property names on one side and values on another. If the value is an array or object just show a button to drill down into the collection. Additional properties and values can be added in a similar fashion as adding rows to a table control inside of a Google doc or MS Word file.

Basically, if I can do everything through drag and drop on my mobile phone, only have to type out the names of things and behavior methods, and never need to look at JSON, then this will be absolutely amazing.

Here's an example of a visual IDE for modeling decision trees. 
https://fxdreema.com
I don't know what tools are used to create that UI but it's quite impressive. I'm not a fan of the code it generates, but it is really good at allowing me to rapidly develop ideas.

Nice example IDEs are MySQL workbench and the Database Designer tools in Visual Studio. It would be really neat to see a UML WYSIWYG designer that works on phones or from a web page and generates what you're already generating. Somewhere inside the database modeling tools and the fxdreema web thing there's probably an idea and a real way of making a great new tool.

This is something I've thought about doing before, I've just never seen anyone else psyched about making functional code from diagrams. I think it's the future of programming because it shortcuts the learning curve and gets directly to visualizing object orientation, which is what we all do anyway. Just a good visual modeler and a few lessons in the most basic programming could probably allow fifth graders to create powerful applications. Blockly sort of did that, but it's not quite what this here could be.

Your project is interesting.

**Comment by ecarriou** [comment]
You are right, programming with diagrams, workflows is the next big step in IDE. In fact I had this idea when I create System Designer, but before going there, I have preferred to start with a simple and light IDE.

One of the great feature I like in an IDE is to design and code only with a keyboard, I did it for System Designer, but It will be less easier I think with a diagramming tool. I have tested some of them and even if they are very powerful, they have poor usability in general. You often do not know how to do the task you want. The more power you give to the user, the more freedom they have and the more chance they have to be lost if you do not focus on usability.

However I see now more and more tools that focus first on usability and then on feature to help programmers (like Swift Playgrounds from Apple) and that is the direction to go.

I will for sure add a more complex UI to drag/drop models and create attributes/links/collections without editing a file, but I do not know yet if it will be for the current project or a new one (with the same foundation).

Anyway thank you for you ideas, I will have a look on fxdreema and check how the UI is.

**Comment by legosteve** [comment]
I read this issue awhile ago but missed the link to fxdreema.com. That is a Sweet Tool. While I wasn’t thinking something that expansive. It would be nice to have the option of editing the schema through a GUI.  Even if it is the same work flow with out any additional auto linking, (although that would be nice). Having the ability to edit as a simple table would be nice.  

Adding an additional page in the page like you have done with the the behaviors editor with the action and behaviors pages.

**Comment by ecarriou** [comment]
Thank you for your feedback! What I have actually in mind is closed to what you have describe. It will be direct edition of the schema/model from the UI. 

For example, instead of going to the MSON file, we edit the property we are on. It will expose less customizations from that mode, but it will be easier to use for beginners.

**Comment by legosteve** [comment]
That sounds great!
